src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs

```diff
@@ -43,14 +43,18 @@ internal abstract class AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer<TSynt
         customTags: [.. DiagnosticCustomTags.Microsoft, EnforceOnBuild.Never.ToCustomTag()]);
 #pragma warning restore RS0030 // Do not used banned APIs
 
+    private static readonly ImmutableArray<DiagnosticDescriptor> s_compilationAnalysisDescriptors = [s_enableGenerateDocumentationFileIdDescriptor];
+
     private readonly DiagnosticDescriptor _classificationIdDescriptor;
     private readonly DiagnosticDescriptor _generatedCodeClassificationIdDescriptor;
+    private readonly ImmutableArray<DiagnosticDescriptor> _semanticModelAnalysisDescriptors;
```

Comment:
i feel like both arrays need docs.  Unclear why we need all of them, and what purpose they serve.

Reply:
➡️ Now documented